src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -940,8 +955,8 @@ private Expression ParseMethodGroupInvocation(MethodGroupExpression methodGroup,
 				throw CreateParseException(errorPos, ErrorMessages.AmbiguousDelegateInvocation);
 
 			var applicableMethod = applicableMethods[0];
-			var delegateExp = Expression.Constant(methodGroup.Overloads.Single(_ => _.Method == applicableMethod.MethodBase));
-			return Expression.Invoke(delegateExp, applicableMethod.PromotedParameters);
+			var usedDeledate = candidates.Single(_ => new[] { _.Method }.Concat(_.InvokeMethods).Any(m => m == applicableMethod.MethodBase)).Delegate;
```

Comment:
Calling `Single` is safe? I mean there aren't cases where an exception can be throw? Maybe if there are more than 1 match?

Reply:
I believe it's safe, because at this point there's a single `applicableMethod`, which comes from one of the delegate. There can be only one matching delegate, because otherwise there would be multiple `applicableMethod`.